src/views/Predictions/hooks/useOnNextRound.ts

```diff
@@ -2,29 +2,43 @@ import { useEffect } from 'react'
 import { useWeb3React } from '@web3-react/core'
 import usePreviousValue from 'hooks/usePreviousValue'
 import { useAppDispatch } from 'state'
-import { useGetCurrentEpoch, useGetSortedRounds } from 'state/predictions/hooks'
+import orderBy from 'lodash/orderBy'
+import { useGetCurrentEpoch, useGetRounds } from 'state/predictions/hooks'
 import useSwiper from './useSwiper'
 
 /**
  * Hooks for actions to be performed when the round changes
  */
 const useOnNextRound = () => {
-  const currentEpoch = useGetCurrentEpoch()
-  const rounds = useGetSortedRounds()
   const { account } = useWeb3React()
-  const previousEpoch = usePreviousValue(currentEpoch)
-  const { swiper } = useSwiper()
   const dispatch = useAppDispatch()
+  const { swiper } = useSwiper()
+  const currentEpoch = useGetCurrentEpoch()
+  const rounds = useGetRounds()
+  const roundsEpochsString = JSON.stringify(
+    Object.keys(rounds)
+      .map((roundKey) => parseInt(roundKey))
+      .sort((a, b) => a - b),
+  )
+  const previousEpoch = usePreviousValue(currentEpoch)
+  const previousRoundsEpochsString = usePreviousValue(roundsEpochsString)
 
   useEffect(() => {
-    if (swiper && currentEpoch !== undefined && previousEpoch !== undefined && currentEpoch !== previousEpoch) {
-      const currentEpochIndex = rounds.findIndex((round) => round.epoch === currentEpoch)
+    if (
+      swiper &&
+      currentEpoch !== undefined &&
+      previousEpoch !== undefined &&
+      (currentEpoch !== previousEpoch || roundsEpochsString !== previousRoundsEpochsString)
+    ) {
+      const currentEpochIndex = orderBy(Object.values(rounds), ['epoch'], ['asc']).findIndex(
```

Comment:
Am I missing something here ?

I don't see the optimization value when adding more logic in this hook.

Reply:
If we don't put that extra roundsEpochsString check, swiper cannot jump on the new current card after the oldest one is removed. (Current epoch is changed before card removal and there would be no extra trigger to run this effect again)